app/frontend/src/utils/transformUtils.js

```diff
@@ -64,3 +65,147 @@ export function attachAttributesToLinks(formSchemaComponents) {
     }
   });
 }
+
+
+/**
+ * This function could be removed from frontend in the next commits
+ * @function isEligibleLateSubmission
+ * Get All possible dates in given period with Term and Interval
+ *
+ * @param {Object[]} date An object of Moment JS date
+ * @param {Integer} term An integer of number of Days/Weeks OR Years
+ * @param {String} interval A string of days,Weeks,months
+ * @returns {Boolean} Return true if form is available for late submission
+ */
+export function isEligibleLateSubmission(date,term,interval){
+  var gracePeriodDate = moment(date,'YYYY-MM-DD HH:mm:ss').add(term,interval).format('YYYY-MM-DD HH:mm:ss');
+  var isBetweenClosrAndGraceDate = moment().isBetween(date, gracePeriodDate);
+  return isBetweenClosrAndGraceDate;
+}
+
+/**
+ * @function getAvailableDates
+ * Get All possible dates in given period with Term and Interval
+ *
+ * @param {Integer} keepAliveFor An integer for number of days
+ * @param {String} keepAliveForInterval A string of days,Weeks,months
+ * @param {Object[]} substartDate An object of Moment JS date
```

Comment:
Why does the parameter say substartDate but the actual declaration is submstartDate?

Reply:
This has turned into `substartDate` which still doesn't match the function declaration parameter `formStartDate`